### DISCUSSION TITLE
Accessing Greensock's GSAP animation libray from the template

### DISCUSSION THREAD

**Discussion by jamesgrubb** [alpinejs/alpine#813] [1. Help] [answered]
Hello Im trying to animate an element created from an Alpine template using GSAP. In react you would use the `useRef` hook to access the DOM element. Not quite sure how to do so with Alpine.
[codepen](https://codepen.io/limitedunlimited/pen/gOMbLxL?editors=1111)
Thanks in advance

**Comment by HugoDF** [comment, marked as answer]
Since the SVG's are rendered in a `x-for`, you'll need to wait until the first Alpine.js render (using `$nextTick`)

I've also seen that you're trying to use the `x-ref` for each of the svg's which isn't going to work properly so I updated that to be `x-ref="svg-0"`, `x-ref="svg-1"` (for illustraction purposes, in your example you probably want to use the `lesson.title`?).

In any case, here's a modified version of your codepen with your GSAP selection fixed: https://codepen.io/hugodf/pen/pobvQoQ?editors=1111

**Reply by jamesgrubb** [reply]
Many thanks, I have not worked with $nextTick yet so it helps to understand it in this context

**Comment by jamesgrubb** [comment]
This throws up a couple of other questions highlighting my lack of javascript knowledge. 

- Can I access the svg innards? ['svg-0' path]
- How do Iterate over the svg so I can chain animations?

**Reply by HugoDF** [reply]
`$refs['svg-0'].querySelector('path')` and so on?

How do you mean iterating?

**Reply by jamesgrubb** [reply]
I'm trying to guess what the equivalent of svg = document.quesrySelectorAll($refs.svg) in Alpine
gsap.to(svg,{x:30,stagger:0.1} // 0.1 seconds between when each "svg" element starts animating

**Reply by HugoDF** [reply]
Yeah there's no querySelectorAll equivalent with refs.

In this case you might actually be better off doing `$el.querySelectorAll('your-selector')`

**Reply by jamesgrubb** [reply]
Ah, perfect, thanks for the explanation, and on a Sunday! Much appreciated.
